File: leet-code/two_pointers/125_is_palindrome.cs
public class Solution
{
    public bool IsPalindrome(string s)
    {
        int start = 0, end = s.Length - 1;
        while(start < end)
        {
            if (!char.IsLetterOrDigit(s[start]))
            {
                start++;
                continue;
            }
            if (!char.IsLetterOrDigit(s[end]))
            {
                end--;
                continue;
            }
            if (char.ToLower(s[start]) != char.ToLower(s[end])) return false;
            start++;
            end--;
        }
        return true;
    }
}

File: leet-code/two_pointers/125_is_palindrome.go
package two_pointers

func IsPalindrome(s string) bool {
	start, end := 0, len(s) - 1
	for start < end {
		if !isCharacter(s[start]) {
			start++
			continue
		}
		if !isCharacter(s[end]) {
			end--
			continue
		}
		if toLower(s[start]) != toLower(s[end]) {
			return false
		}
		start++
		end--
	}
	return true
}

func isCharacter(character byte) bool {
	return (character >= '0' && character <= '9') || (character >= 'A' && character <= 'Z') || (character >= 'a' && character <= 'z')
}

func toLower(character byte) byte {
	if character >= 'A' && character <= 'Z' {
		return character + 32
	}
	return character
}